### PR TITLE
CRT HTTP/1 GA - Surface Area Updates - readBufferSizeInBytes takes long

### PR DESCRIPTION
**Pull request by dave-fn**
PR in support for CRT HTTP/1 GA - Surface Area Updates. This PR modifies the `readBufferSizeInBytes` to take a `long`, not `int` parameter.

## Modifications
Updated method and cascaded changes into `CrtRequestContext` and `CrtRequestBodyAdapter`.

## Testing
No new tests added.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java [hunk 1]
```diff
@@ -325,7 +325,7 @@ public interface Builder extends SdkAsyncHttpClient.Builder<AwsCrtAsyncHttpClien
          *
          * TODO: This is also used for the write buffer size. Should we rename it?
          */
-        Builder readBufferSizeInBytes(Integer readBufferSize);
+        Builder readBufferSizeInBytes(Long readBufferSize);
```
Comment (zoewangg) [inline http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:328]: Can we add documentation for this behavior? Something like: the max allowed read buffer size is Integer.MAX.
While we are at it, can we remove the TODO at line 326?

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java [hunk 2]
```diff
@@ -207,7 +207,7 @@ private HttpClientConnectionManager createConnectionPool(URI uri) {
                 .withSocketOptions(socketOptions)
                 .withTlsContext(tlsContext)
                 .withUri(uri)
-                .withWindowSize(readBufferSize)
+                .withWindowSize((int) readBufferSize)
```
[the remark this answers is not in the document]
Comment (dave-fn) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:210]: Yeah. Thanks for showing me this utility.

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java (hunk 1 above)
Comment (dave-fn) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:328]: Done.
On a similar note, what about the comment at line 69? Do we still want to keep it?

Comment (zoewangg) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:328]: Yeah, we can leave it as is now, and I'll create a PR to remove all of PREVIEW wordings when it's ready

**Review by zoewangg — APPROVED** [review]

**Pull request merged**